src/main/groovy/net/nemerosa/versioning/VersioningExtension.groovy

```diff
@@ -151,6 +153,27 @@ class VersioningExtension {
      */
     String lastTagPattern = /(\d+)$/
 
+    /**
+     * Compute version code
+     *
+     * Closure that compute versionCode from <i>scmInfo</i>, <i>versionReleaseType</i>, <i>versionBranchId</i>,
+     * <i>versionFull</i>, <i>versionBase</i> and <i>versionDisplay</i>
+     *
+     * By default it tries to find this pattern in display : "\d{2}.\d{2}.\d{2}". Version code is computed with
+     * this algo : code = group(1) * 10000 + group(2) * 100 + group(3)
+     */
+    Closure<Integer> computeVersionCode = { SCMInfo scmInfo, String versionReleaseType, String versionBranchId,
+                                            String versionFull, String versionBase, String versionDisplay ->
+        String code = getVersionCodeFromName(versionDisplay)
+        int ret = 0
+        try {
+            ret = code.toInteger()
+        } catch (NumberFormatException ignore) {
+            println("Version code is ${code}, use $ret as version code")
```

Comment:
Is that a desired behavior? Should be `0` used as version code when the code returned by `getVersionCodeFromName` cannot be parsed?

Reply:
We must have a default number as a versionCode because it is a int. I think that 0 is the most neutral number in this case. 
Also, my opinion is that is versionCode cannot be computed, then this info may not be used.
Nevertheless, I also though about putting 1 in this variable when versionCode cannot be computed. What do you prefer ?

Reply:
In my next commit, I have made this default number configurable